---
title: "Analyse_génotypes_FruitSelGen"
author: "Mariem NSIBI"
date: "July 3, 2017"
output: word_document
---

## Packages
```{r}
source("https://bioconductor.org/biocLite.R")
biocLite()
biocLite("VariantAnnotation")
library(VariantAnnotation)
library(rutilstimflutre)
```

##export from vcf to genoClasses


```{r}
vcf2genoClasses(vcf.file="FruitSelGen-apricot-PRPER-Lovell-v2-avignon_GoxMo_raw-snps_gatk-filter.vcf.gz", genome="PRPER-Lovell-v2",
                gclasses.file="FruitSelGen-apricot-PRPER-Lovell-v2-avignon_GoxMo_raw-snps_gatk-filter_gclasses.tsv.gz",
                ca.file="FruitSelGen-apricot-PRPER-Lovell-v2-avignon_GoxMo_raw-snps_gatk-filter_coords-alleles.tsv.gz",
                yieldSize=4, verbose=1)
```
# Filter cases of segregation distortion
```{r}
gclasses <- read.table("FruitSelGen-apricot-PRPER-Lovell-v2-avignon_GoxMo_raw-snps_gatk-filter_gclasses.tsv", stringsAsFactors=FALSE, check.names=FALSE)
dim(gclasses)
gclasses[1:3, 1:5]
```
#identification of SNPs which genotypes show a distortion of segregation
#segregation distortion 
```{r}
idx.par1 <- grep("Goldrich", colnames(gclasses))
idx.par2 <- grep("Moniqui", colnames(gclasses))
idx.offs <- setdiff(1:ncol(gclasses), c(idx.par1, idx.par2))
in.gc <- gclasses[, c(idx.par1, idx.par2, idx.offs)]
dim(in.gc)
jm <- genoClasses2JoinMap(x=in.gc, na.string=NA,thresh.counts=round(0.1*(ncol(in.gc)-2)),thresh.na=round(0.3*(ncol(in.gc)-2)))
dim(jm)
head(jm)
write.table(jm, file=gzfile("FruitSelGen-apricot-PRPER-Lovell-v2-avignon_GoxMo_joinmap.tsv.gz"), quote=FALSE, sep="\t")
```

```{r}
out.chi2 <- filterSegreg(x=jm[, -(1:8)], return.counts=TRUE)
dim(out.chi2)
head(out.chi2[! is.na(out.chi2$pvalue),])
p2f <- "FruitSelGen-apricot-PRPER-Lovell-v2-avignon_GoxMo_segreg-chi2.tsv.gz"
write.table(out.chi2, file=gzfile(p2f),
            quote=FALSE, sep="\t")
plotHistPval(pvalues=out.chi2$pvalue, verbose=1)
install_github("qvalue")
library(qvalue)
qqplotPval(pvalues=out.chi2$pvalue, verbose=1)
```
# Export the filtered genotypic data
```{r}
p2f <- "FruitSelGen-apricot-PRPER-Lovell-v2-avignon_GoxMo_segreg-filtered.loc.gz"
tmp <- out.chi2[! is.na(out.chi2$pvalue),]
length(snps.tokeep <- rownames(tmp[tmp$pvalue.bonf > 0.05,]))
cohort<-"GoxMo"
writeSegregJoinMap(pop.name=cohort, pop.type="CP",locus=snps.tokeep,segregs=jm[snps.tokeep, "seg"],phases=rep("{?}", length(snps.tokeep)), classifs=rep("NA", length(snps.tokeep)),genos=jm[snps.tokeep, -c(1:9)],file=p2f)
```
#subset the initial VCF to only keep the same SNPs:
```{r}
filterVariantCalls("FruitSelGen-apricot-PRPER-Lovell-v2-avignon_GoxMo_raw-snps_gatk-filter.vcf.gz", genome="Lovell-v2",
                   yieldSize=1000,
                   variants.tokeep=snps.tokeep,
                   out.file=paste0("FruitSelGen-apricot-PRPER-Lovell-v2-avignon_GoxMo_segreg-filtered.vcf.gz"))

```
# Appendix

```{r info}
t1 <- proc.time(); t1 - t0
print(sessionInfo(), locale=FALSE)
```